CovertInfiltration/Src/CovertInfiltration/Classes/X2StrategyElement_InfiltrationRewards.uc

```diff
@@ -418,16 +418,19 @@ static function GiveTechInspireReward(XComGameState NewGameState, XComGameState_
 
 	// Adjust Tech's time reduction value
 	TechState = XComGameState_Tech(NewGameState.ModifyStateObject(class'XComGameState_Tech', RewardState.RewardObjectReference.ObjectID));
-	TechState.TimeReductionScalar = class'X2StrategyElement_DefaultRewards'.static.GetTechRushReductionScalar();
-
-	// If there is already a project rush it
-	foreach History.IterateByClassType(class'XComGameState_HeadquartersProjectResearch', ProjectState)
+	if(TechState != None && TechState.TimesResearched > 0)
```

Comment:
This is confusing to me. What you are saying is "speed up research only if it was researched before". I think a better way to handle this would be to save `TimesResearched` to reward state in the generate/set functions and then compare it here to the saved value.

You could perhaps use the `var int Quantity` on the reward state to avoid creating more state variables

Reply:
Oh whoops that's supposed to be ` == 0`